### PR TITLE
Fold flash setting into subsequent redirect

### PR DESCRIPTION
**Pull request by tommotorefi**
I'm not sure if this is the best way to write this rule; corrections welcome!

### CODE REVIEW

File: lib/rails/redirect_with_flash.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  redirect_to root_path, flash: {error: "huzzah"}
+
+  EOS
+  within_file 'app/controllers/**/*rb' do
+    within_node type: 'def' do
+      line = nil
+      msg = nil
+      remover_action = nil
+      flash_type = nil
+      with_node type: 'send', receiver: 'flash', arguments: {size: 2, first: %r{:notice|:error}, last: {type: :str}} do
```
Comment (flyerhzm) [inline lib/rails/redirect_with_flash.rb]: why do you just check `notice` and `error`? it could be `alert` or anything else

**Comment by tommotorefi** [comment]
@flyerhzm good point, I've pushed a new version that handles any type, and also added another spec for a case where the rule does not make a change to the source.

File: lib/rails/redirect_with_flash.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+Synvert::Rewriter.new 'rails', 'redirect_with_notice' do
```
Comment (flyerhzm) [inline lib/rails/redirect_with_flash.rb]: snippet name should be `redirect_with_flash`

**Comment by tommotorefi** [comment]
@flyerhzm woops right on, fixed.

**Pull request merged**